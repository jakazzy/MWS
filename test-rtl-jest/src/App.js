import React from 'react';


function App( { text }) {
  return (
      <header>
       <h2 data-testid="h2tag" className="fancy-h2"> { text }</h2> 
      </header>
  );
}

export default App;
