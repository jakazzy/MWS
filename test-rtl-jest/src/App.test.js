import React from 'react';
import 'jest-dom/extend-expect'
import { render, cleanup} from '@testing-library/react'
import App from './App';

afterEach(cleanup);

it('renders without crashing', () => {
  const { asFragment} = render( <App text='Hello!'/>);
  expect(asFragment()).toMatchSnapshot();
});

it('inserts text in h2',() =>{
  const {getByTestId, getByText } = render(<App text='Hello!'/>);
  expect(getByTestId('h2tag')).toHaveTextContent('Hello!');
  expect(getByText('Hello!')).toHaveClass('fancy-h2');
})