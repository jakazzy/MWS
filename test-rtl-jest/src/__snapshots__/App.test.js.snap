// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <header>
    <h2
      class="fancy-h2"
      data-testid="h2tag"
    >
       Hello!
    </h2>
  </header>
</DocumentFragment>
`;
